### PR TITLE
network: fixes several issues in IPv4 DAD for DHCP4

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/network/networkd-network.h [hunk 1]
```diff
@@ -123,7 +122,6 @@ struct Network {
         bool dhcp_send_release;
         bool dhcp_send_decline;
         DHCPUseDomains dhcp_use_domains;
-        sd_ipv4acd *dhcp_acd;
```
Comment (keszybz) [inline src/network/networkd-network.h:126]: Ooops, that must have caused a lot of fun side-effects if anyone used the same .network file with two interfaces...

File: src/network/networkd-link.c [hunk 2]
```diff
@@ -623,6 +623,12 @@ int link_stop_clients(Link *link, bool may_keep_dhcp) {
                         r = log_link_warning_errno(link, k, "Could not stop DHCPv4 client: %m");
         }
 
+        if (link->dhcp_acd) {
+                k = sd_ipv4acd_stop(link->dhcp_acd);
```
Comment (keszybz) [inline src/network/networkd-link.c:627]: Hmm, maybe we should make `sd_ipv4acd_stop()` do nothing on NULL?

**Review by keszybz — COMMENTED** [review]
Looks all reasonable.

File: src/network/networkd-link.c (hunk 2 above)
Comment (yuwata) [reply src/network/networkd-link.c:627]: Year, I will create another PR later.

**Pull request merged**